slm_porosity/heat_transfer.f90

```diff
@@ -704,6 +716,42 @@ SUBROUTINE user_read_input ()
     CALL check_user_input
   END SUBROUTINE user_read_input
 
+  SUBROUTINE read_laser_trajectory (trajectory_file_)
+    IMPLICIT NONE
+    CHARACTER(LEN=*), INTENT(IN) :: trajectory_file_
+    INTEGER  :: stat
+    INTEGER  :: line
+    REAL(pr) :: buf(6)
+
+    line = 0
+    OPEN(042, FILE=trajectory_file_, ACTION='read')
+    DO
+      READ(042, *, iostat=stat) buf
+      IF (stat.NE.0) EXIT
+      line = line + 1
+    END DO
+    ALLOCATE(trajectory(line))
+    CLOSE(042)
+    line = 0
+    OPEN(042, FILE=trajectory_file_, ACTION='read')
+    DO
+      READ(042, *, iostat=stat) buf
+      IF (stat.NE.0) EXIT
+      line = line + 1
+      trajectory(line)%position = buf(1:3)
```

Comment:
Check the length of the buf. It can be 2/4/6. Then we can use default zero values. For example, for `z` axis. See the last comments.